### PR TITLE
Fix reference type fields in Firestore throwing 500

### PR DESCRIPTION
**Pull request by sharat87**

### CODE REVIEW

File: app/server/appsmith-plugins/firestorePlugin/src/main/java/com/external/plugins/FirestorePlugin.java [hunk 1]
```diff
@@ -388,13 +396,15 @@ private Object resultToMap(Object objResult) throws AppsmithPluginException {
                 resultMap.put("path", documentReference.getPath());
                 return resultMap;
 
-            } else {
+            } else if (isRoot) {
```
Comment (trishaanand) [inline app/server/appsmith-plugins/firestorePlugin/src/main/java/com/external/plugins/FirestorePlugin.java:399]: Do we not need another else which handles the scenario of not finding the objResult of any of the previous types and is not a root?

Comment (sharat87) [reply app/server/appsmith-plugins/firestorePlugin/src/main/java/com/external/plugins/FirestorePlugin.java:399]: Yeah that's where we return `objResult` itself on line 407 if there's no match in the above `if` conditions.

**Review by trishaanand — APPROVED** [review]

**Comment by tazmaniax** [comment]
Hi, do you now when this will be live on the cloud version? thx

**Comment by trishaanand** [comment]
> Hi @trishaanand, do you now when this will be live on the cloud version? thx

@tazmaniax : We should merge to master either tonight or tomorrow morning. We are currently blocked by github services seeing some degradation.

**Pull request merged**